File: q2090/q2090.py
class Solution:
    def getAverages(self, nums: List[int], k: int) -> List[int]:
        if k == 0:
            return nums

        n = len(nums)
        averages = [-1] * n
        if n < 2 * k + 1:
            return averages

        currentSum = sum(nums[: 2 * k + 1])
        averages[k] = currentSum // (2 * k + 1)

        for j in range(k + 1, n - k):
            l = j - k - 1
            r = j + k
            currentSum = currentSum - nums[l] + nums[r]
            averages[j] = currentSum // (2 * k + 1)

        return averages

File: q2090/q2090.cpp
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> getAverages(vector<int> &nums, int k)
    {
        if (k == 0)
            return nums;

        // By default, the positions are filled with -1
        int n = nums.size();
        vector<int> averages(n, -1);

        if (n < 2 * k + 1)
            return averages;

        // Compuet the first valid window sum
        long currentSum = 0;
        for (int i = 0; i <= 2 * k; i++)
        {
            currentSum += nums[i];
        }
        averages[k] = currentSum / (2 * k + 1);

        // At every index, subtract from left and add from the right. Then recompute the average
        for (int j = k + 1; j < n - k; j++)
        {
            int l = j - k - 1;
            int r = j + k;
            currentSum += nums[r];
            currentSum -= nums[l];
            averages[j] = currentSum / (2 * k + 1);
        }

        return averages;
    }
};